src/ChangeStream.php

```diff
@@ -214,9 +222,7 @@ private function onIteration($isNext)
             return;
         }
 
-        /* Increment the key if the iteration event was a call to next() and we
-         * have already advanced past the first result. */
-        if ($isNext && $this->hasAdvanced) {
+        if ($incrementKey) {
```

Comment:
I contemplated renaming this to `$incrementKeyIfValid`, but adding a comment above the early return for `!$this->valid()` seemed preferable.

Reply:
I think the comment makes this clear.